File: we-mov-backend/src/main/java/com/pl/wemovbackend/entities/Customer.java
package com.pl.wemovbackend.entities;

import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "customer")
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long customerId;
    @NonNull
    private String customerName;
    @Column(unique = true)
    private String mobile;
    @Column(unique = true)
    private String email;
}

File: we_mov_mobile/lib/models/customer.dart
class Customer {
  int? customerId;
  String customerName;
  String mobile;
  String email;

  Customer({
  this.customerId,
  required this.customerName,
  required this.mobile,
  required this.email,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    print(json);

    return Customer(
  customerId: json['customerId'],
  customerName: json['customerName'].toString(),
  mobile: json['mobile'].toString(),
  email: json['email'].toString(),
  );
  }

  Map<String, dynamic> toJson() {
    return {
      "customerId": this.customerId,
      "customerName": this.customerName,
      "mobile": this.mobile,
      "email": this.email,
    };
  }

  //
}
